lib/src/model/model_node.dart

```diff
@@ -36,26 +57,13 @@ class ModelNode {
   }
 
   late final String sourceCode = () {
-    var enclosingSourceNode = _sourceNode;
-    if (enclosingSourceNode == null) {
+    if (_sourceEnd < 0 || _sourceOffset < 0) {
       return '';
     }
 
-    /// Get a node higher up the syntax tree that includes the semicolon.
-    /// In this case, it is either a [FieldDeclaration] or
-    /// [TopLevelVariableDeclaration]. (#2401)
-    if (enclosingSourceNode is VariableDeclaration) {
-      enclosingSourceNode = enclosingSourceNode.parent!.parent;
-      assert(enclosingSourceNode is FieldDeclaration ||
-          enclosingSourceNode is TopLevelVariableDeclaration);
-    }
-
-    var sourceEnd = enclosingSourceNode!.end;
-    var sourceOffset = enclosingSourceNode.offset;
-
     var contents = model_utils.getFileContentsFor(element, resourceProvider)!;
```

Comment:
You should not do this, the analyzer already gives `ResolvedUnitResult.content`.

Reply:
I agree 100%. This is shocking that we re-read everything.